### DISCUSSION TITLE
Set scoring plane position

### DISCUSSION THREAD

**Discussion by tairozero** [nrc-cnrc/EGSnrc#985] [Questions]
I want to create general head-tube for measure backscatter factor in diagnostic radiology and already have accelerator and egsphant CT phantom file for simulation. However, I don’t sure that I have correct understanding in some condition, so please help me checked it.

I decided to used isource 9 (Beam treatment head simulation used as source, incident from any angle). If I want to get the dose at the surface of phantom and 100 cm Focal-surface distance, I must set the “distance from source to iso-center (cm)” at the surface of the phantom with the same position of scoring plane in beamnrc setting and move my focal spot in beamnrc to correspond with 100 cm of FSD?

Even if, my phantom’s position is in the -y direction. Must I have set the source in -y direction too?

**Comment by rtownson** [comment]
For a beam aimed along -y, use this:
```
./coordinateTransform.py 180 0 0
('Gantry, couch, collimator:', 180.0, 0.0, 0.0)
('Theta, phi, phicol:', 90.0, 90.0, 270.0)
```
The `dsource` parameter in DOSXYZnrc corresponds for source 9 to the distance from the isocenter to the source scoring plane in BEAMnrc. So let's say you BEAMnrc your scoring plane is at z=50cm, your isocenter is at 0,0,0, and your phantom surface is at y=+10. Then use the theta/phi/phicol from above, with dsource=60. This means your FSD is 100, and FAD (focal to axis distance) is 110.

Duplicate post: https://www.reddit.com/r/EGSnrc/comments/12ml745/check_infirmation/

**Reply by tairozero** [reply]
Where must I have to add the three code that you mention? And myisocenter maybe 0,0,90

**Reply by rtownson** [reply]
The 'code' was just the results of running the script from this paper, that I have shared with you before: https://pubmed.ncbi.nlm.nih.gov/23175216/

So you can just use the angles from it.

**Reply by tairozero** [reply]
So if i must to set FSD is 100 cm, my isocenter is at 0,0,-90 and my phantom surface is at y=-5. Then I must to set dsource=95 and my scoring plane is at z=100?

**Reply by rtownson** [reply]
If your scoring plane is at z=100, then you need to place it right on the surface of your phantom. The distance along y from your isocenter to the surface is -5cm=dsource. Then you just need to make sure the angles are right, so the beam is indeed aimed along -y. Though is it strange to me to start your source past the isocenter, if indeed the beam is aimed along -y. I'm not sure if this will cause confusion with the angles, so if it doesn't look right you could consider adjusting the beamnrc scoring plane to be a bit further from the surface.

**Reply by tairozero** [reply]
So, should I try to set the scoring plane at 95 cm? And set the angle to 90 (theta), 90 (phi) and 270 (collimator angle) and 90 for dsource? I just have some confuse about the degree can you help me check the parameter setting?

**Reply by rtownson** [reply]
If the scoring plane is at 95, that's at the isocenter, right? So dsource=0 because it's the distance from the isocenter to the scoring plane. Yes, those angles will make a beam along -y. You can test the angles with very fast simulations in the example 16MVp_h2o_phantom_beamsource_example.egsinp (just turn on the option for it to output an egsphant so you can view it).

**Reply by tairozero** [reply]
My isocenter in the phantom is 0,0,-90 but I want to set the scoring plane at the surface phantom at y = -5

**Reply by rtownson** [reply]
Yes, this is consistent with what I described. The beam along -y, scoring plane at z=95 from the focal spot, which after being rotated about the isocenter to aim along -y you want at the isocenter. Then there's 5cm along -y until the surface, so the total distance is 95+5=FSD. Thus dsource=0 because you've already accounted for 95cm in the BEAMnrc simulation, and then 5cm between the isocenter and y=-5 along y.

**Reply by tairozero** [reply]
Ok, I’ll try it.
Thank you so much.

**Reply by tairozero** [reply]
![Screenshot from 2023-04-19 12-40-40](https://user-images.githubusercontent.com/130958740/232961280-72fcf7b6-6905-40ab-a5ff-33907dce0392.png)

In my VICTORIA the -y direction is upper and +y direction is lower. in that case if I set the position of scoring plane is 95 cm and dsouce = 0 my FSD is not 90?

**Reply by rtownson** [reply]
Oh, you want the beam aimed along +y then, not -y. So the angles should be 90, 270, 270. Now if the scoring plane is at z=95 in BEAMnrc, you'll need 5cm more along the y axis before hitting the surface at y=-5. This means you want dsource=10cm.

**Reply by tairozero** [reply]
Thank you for your answer

**Reply by tairozero** [reply]
I have a question. when I set the scoring plane in the beamnrc with half-width of outer square boundary (cm) is 5 cm. That means I got the scoring plane 10x10 cm right?

**Reply by rtownson** [reply]
Are you talking about the half-width of the outer square boundary of a CM, like SLABS? This is not just to do with the scoring plane, so I will explain. This is the outer boundary of the CM material (which yes, would be 10x10cm if you set it to 5). Be careful setting this small because all particles outside that boundary are lost, so if you have one narrow CM in the middle of you linac it will cut off all particles outside that boundary when they go through that CM. If you change this parameter on the CM for your scoring plane, then yes it will control the boundary of your scoring plane. Use beamdp_gui with a scatter plot to observe the effect.

**Reply by tairozero** [reply]
I got the 3ddose file and tried to read by statdose, but it showed that my dimension is greater than allow limit (my number voxel in x y z direction is 118 118 and 870, but the limit is 200 200 and 200).

I tried to change the lines 213 and 214 ($MAXVOXZ and $MAXVOXEL) in statdose.mortran to 1000 and 1500, respectively and re-complie again.
But the errored occur and showed like the sentence below

Mortran compiling statdose sources ...
mortran3.exe: using following I/O units
  raw/hex data file: /home/asada/EGSnrc/HEN_HOUSE/bin/linux64/mortran3.dat
  command line files: 
  -> appending /home/asada/EGSnrc/HEN_HOUSE/lib/linux64/machine.macros
  -> appending statdose.mortran
  -> appending /home/asada/EGSnrc/HEN_HOUSE/lib/linux64/machine.mortran
  -> appending /home/asada/EGSnrc/HEN_HOUSE/utils/xvgrplot.mortran
  -> appending /home/asada/EGSnrc/HEN_HOUSE/src/lnblnk1.mortran
  generated mortjob.mortran
  fortran output: statdose_linux64.f
  mortlst output: statdose_linux64.mortlst

  Mortran compiling ...  OK
Fortran compiling statdose_linux64.f using flags -O2 -mtune=native -fPIC 
/tmp/ccj6LWSE.o: in function `lettercount_':
statdose_linux64.f:(.text+0x1f0): relocation truncated to fit: R_X86_64_PC32 against `.bss'
statdose_linux64.f:(.text+0x218): relocation truncated to fit: R_X86_64_PC32 against `.bss'
/usr/bin/ld: failed to convert GOTPCREL relocation; relink with --no-relax
collect2: error: ld returned 1 exit status
make: *** [Makefile:40: /home/asada/EGSnrc/HEN_HOUSE/bin/linux64/statdose] Error 1

What should I do?

And this is my 3ddose file when I tried to see the result in VICTORIA
![Screenshot from 2023-04-22 15-03-12](https://user-images.githubusercontent.com/130958740/233892747-0b5a2125-90cc-4cde-a43d-28627131c2e2.png)


I don't sure is it normal result or not? Why I have may gray color doses in air?

**Reply by rtownson** [reply]
To get statdose (and dosxyznrc) to compile with so many voxels, you'll need to adjust compiler flags, and be sure you have a lot of RAM available. Expect to need about RAM equal to about X\*Y\*Z\*8bytes\*5, where X is your maximum number of voxels along X, etc.

See here: https://www.reddit.com/r/EGSnrc/comments/jkcv4e/increase_the_maximun_number_of_voxels_in_xyz/

The grey doses I'm not sure, but just check the dose profile, does it make sense?

**Reply by tairozero** [reply]
How can I find the flags? Is in statdose.mortran?

**Reply by rtownson** [reply]
In `omega/progs/statdose/Makefile` change
```
@$(F77) $(FOPT) $(FCFLAGS) $(FOUT)$@ statdose_$(my_machine).f
```
to
```
@$(F77) $(FOPT) $(FCFLAGS) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -mcmodel=large $(FOUT)$@ statdose_$(my_machine).f
```
and trying recompiling statdose. I assume you already had to do this for ctcreate and dosxyznrc in order to get them to compile with so many voxels.

**Reply by tairozero** [reply]
I didn’t change in ctcreate and dosxyznrc yet, should I change too?

**Reply by tairozero** [reply]
I've tried to change the line follow your mention, Unfortunately, my RAM is not have space enough to compile it. However, I'll try to use external resources like 3ddose_tool or Profeel

**Comment by tairozero** [comment]
Yes, I set the lowest SLAB as scoring plane to use as Field size with the 10x10 cm to 40x40 cm so, that mean I must set the outer boundary between 5-20 cm right? And after I got the 3ddose file, how I can read the dose in the file?

**Reply by rtownson** [reply]
It's not realistic, but yes it will cut the field. Normally you would use jaws or an applicator to collimate. I assuming the beam is diverging so keep in mind that 10x10 at 50cm is not the same as at 100cm.

Edit: https://github.com/nrc-cnrc/EGSnrc/wiki/Visualizing-or-parsing-3ddose-files

**Reply by tairozero** [reply]
I quite confused. Follow as our last parameter the field size (scoring plane) is 100 cm from the focal spot, so what about 50 cm that you told me? What does it mean?

**Reply by rtownson** [reply]
I was just giving an example, and forgot your last CM is indeed at 100. Then just keep in mind that cutting the field at 100cm may not be the same as the collimation in whatever realistic environment you are trying to model.

**Reply by tairozero** [reply]
Thank you for ypur explain. In this case, I just already set the collimator for the beam divergent.

**Reply by rtownson** [reply]
If your collimator is already set to shape the beam, then don't use a small boundary on the SLAB. You might lose particles that scattered outside of the field size.

**Reply by tairozero** [reply]
So, if I didn’t use the small boundary SLAB. How can I measure the dose on the phantom’s surface?

**Reply by rtownson** [reply]
I don't see how changing the boundary changes this - you would still use a tool like VICTORIA to look at the dose profiles in the 3ddose file from DOSXYZnrc.

**Reply by tairozero** [reply]
I still using VICTORIA to see dose profile, since I can’t use statdose. And I didn’t show the dose result from boundary changing. It’s still simulating.

**Reply by tairozero** [reply]
Or just I should to extend the boundary to cover all of the phantom’s surface?

**Reply by rtownson** [reply]
The boundary of the CM is an artificial cut-off for the radiation field, so you should make it large enough that it does not affect the quantities of interest. The only reason to reduce it is for an accuracy/efficiency trade-off. Ideally, all collimation should be done by realistic objects modelled in your simulation.

**Reply by tairozero** [reply]
Thank you for your information. In this case, Should I change only the half-width of outer square boundary  or both of half-width and thickness of the SLAB?
